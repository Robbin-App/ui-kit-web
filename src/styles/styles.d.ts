import 'styled-components';
import { IBaseRobbinTheme } from './theme';

declare module 'styled-components' {
  export type DefaultTheme = IBaseRobbinTheme;
}
