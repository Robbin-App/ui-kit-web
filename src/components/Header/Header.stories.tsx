import React from 'react';
import { Meta, Story } from '@storybook/react';
import WorkOutlineIcon from '@material-ui/icons/WorkOutline';
import DescriptionOutlinedIcon from '@material-ui/icons/DescriptionOutlined';
import NotificationsNoneOutlinedIcon from '@material-ui/icons/NotificationsNoneOutlined';
import { LogoType } from '../Logo';
import { IHeaderProps, Header } from './index';

export default {
  title: 'Components/Header',
  component: Header,
} as Meta;

const Template: Story<IHeaderProps> = (args) => <Header {...args} />;

export const defaultProps = {
  logoType: LogoType.TalentosNormal,
  navItems: [
    {
      label: 'Oportunidades',
      href: '/',
      // eslint-disable-next-line react/display-name
      renderIcon: () => <WorkOutlineIcon />,
    },
    {
      label: 'Aplicaciones',
      href: '/applications',
      // eslint-disable-next-line react/display-name
      renderIcon: () => <DescriptionOutlinedIcon />,
    },
    {
      label: 'Notificaciones',
      href: '/notifications',
      // eslint-disable-next-line react/display-name
      renderIcon: () => <NotificationsNoneOutlinedIcon />,
    },
    {
      label: 'Editar Perfil',
      href: '/edit-profile',
      // eslint-disable-next-line react/display-name
      renderIcon: () => <NotificationsNoneOutlinedIcon />,
    },
  ],
  pathname: '/',
  userFirstName: 'John',
  userLastName: 'Doe',
};

export const Default = Template.bind({});
Default.args = defaultProps;
