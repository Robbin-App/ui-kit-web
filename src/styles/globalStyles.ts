import { createGlobalStyle, css } from 'styled-components';
import { normalize } from 'styled-normalize';

export const globalStyles = css`
  ${normalize}
  * {
    box-sizing: border-box;
    font-feature-settings: "lnum";
    &:focus {
      outline: 1px solid ${(props) => props.theme.colors.secondary100};
    }
  }
  body {
    font-family: 'Nunito', sans-serif;;
    color: ${(props) => props.theme.colors.text.globalText}
    background: ${(props) => props.theme.colors.greySurface};
    font-size: 16px;
  }
  html,
  body {
    overflow-x: hidden;
    position: relative;
  }
  
  html {
    scroll-behavior: smooth;
  }
  
  input {
    -webkit-appearance: none;
  }

  input:-webkit-autofill,
  input:-webkit-autofill:hover,
  input:-webkit-autofill:focus textarea:-webkit-autofill,
  textarea:-webkit-autofill:hover textarea:-webkit-autofill:focus,
  select:-webkit-autofill,
  select:-webkit-autofill:hover,
  select:-webkit-autofill:focus {
    -webkit-box-shadow: 0 0 0px 1000px #ffffff inset !important;
  }
  
  h1, h2, h3, h4, h5,h6 {
    color: ${(props) => props.theme.colors.text.headingText}
  }
  
  a {
    color: inherit;
  }
`;

export const GlobalStyles = createGlobalStyle`${globalStyles}`;
