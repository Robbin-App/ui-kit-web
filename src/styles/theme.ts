export interface IBaseRobbinTheme {
  colors: {
    primary90: string;
    primary100: string;
    primary200: string;
    secondary90: string;
    secondary100: string;
    secondary200: string;
    error: string;
    text: {
      headingText: string;
      globalText: string;
      disabledText: string;
    };
    border: {
      darkBorder: string;
      lightBorder: string;
    };
    white: string;
    greySurface: string;
  };
  shadows: {
    global: string;
    globalHover: string;
    focus: string;
    active: string;
  };
  fonts: {
    nunito: string;
    weight: {
      light: number;
      regular: number;
      semiBold: number;
      bold: number;
      extraBold: number;
    };
  };
  sizes: {
    containerMaxWidth: number;
    borderRadius: string;
  };
}

export const baseRobbinTheme: IBaseRobbinTheme = {
  colors: {
    white: '#ffffff',
    greySurface: '#FAFAFA',
    primary90: '#1A5589',
    primary100: '#15456F',
    primary200: '#133D63',
    secondary90: '#5081B9',
    secondary100: '#4677AF',
    secondary200: '#426FA4',
    error: '#A62121',
    border: {
      lightBorder: '#E7E6E6',
      darkBorder: '#D6D6D6',
    },
    text: {
      headingText: '#2E2E2E',
      globalText: '#545454',
      disabledText: '#A1A1A1',
    },
  },
  shadows: {
    global: '0px 0px 9px 2px rgba(0, 0, 0, 0.08)',
    globalHover: '0px 0px 9px 2px rgba(0, 0, 0, 0.16)',
    focus: '0px 0px 9px 2px rgba(21, 69, 111, 0.5)',
    active: '0px 0px 9px 2px rgba(21, 69, 111, 0.8)',
  },
  fonts: {
    nunito: `'Nunito', sans-serif`,
    weight: {
      light: 300,
      regular: 400,
      semiBold: 600,
      bold: 700,
      extraBold: 800,
    },
  },
  sizes: {
    containerMaxWidth: 1230,
    borderRadius: '8px',
  },
};
