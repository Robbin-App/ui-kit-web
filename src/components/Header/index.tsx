import React, { ReactElement, useState } from 'react';
import styled from 'styled-components';
import MenuIcon from '@material-ui/icons/Menu';
import CloseIcon from '@material-ui/icons/Close';
import Avatar from '@material-ui/core/Avatar';
import { Logo, LogoType } from '../Logo';
import { container } from '../../styles';
import { HeaderLink } from '../HeaderLink';

export interface INavItems {
  href: string;
  label: string;
  renderIcon(): ReactElement;
}

export interface IHeaderProps {
  /**
   * Logo Type of the Header
   */
  logoType: LogoType;
  /**
   * Navigation Items of the Header
   */
  navItems: INavItems[];
  /**
   * Navigation Items Link Wrapper
   */
  navItemLinkWrapper?(
    props: INavItems,
    children: ReactElement
  ): ReactElement | null;
  /**
   * Current site pathname
   */
  pathname: string;
  /**
   * Logged User First Name
   */
  userFirstName?: string;
  /**
   * Logged User Last Name
   */
  userLastName?: string;
  /**
   * Function to render the right side avatar component
   */
  avatarLinkWrapper?(children: ReactElement): ReactElement | null;
}

/**
 * Basic Header Component
 */
export const Header: React.FunctionComponent<IHeaderProps> = ({
  navItems,
  logoType,
  userLastName,
  userFirstName,
  children,
  pathname,
  avatarLinkWrapper = () => null,
  navItemLinkWrapper = () => null,
}) => {
  const [mobileOpen, setMobileOpen] = useState<boolean>(false);

  return (
    <StyledHeader>
      <InnerContainer>
        <Logo type={logoType} />
        <NavContainer>
          <StyledNav open={mobileOpen}>
            {children}
            {navItems.map((navItem) =>
              navItemLinkWrapper(
                navItem,
                <HeaderLink
                  key={navItem.href}
                  label={navItem.label}
                  renderIcon={navItem.renderIcon}
                  active={pathname === navItem.href}
                />
              )
            )}
          </StyledNav>
        </NavContainer>
        <DesktopAvatar>
          {avatarLinkWrapper(
            <Avatar>
              {userFirstName &&
                userLastName &&
                `${userFirstName.charAt(0)}${userLastName.charAt(0)}`}
            </Avatar>
          )}
        </DesktopAvatar>
        <MobileMenuTrigger
          aria-hidden={mobileOpen}
          onClick={() => setMobileOpen((prev) => !prev)}
          tabIndex={mobileOpen ? 0 : -1}
          aria-label={mobileOpen ? 'cerrar menu' : 'abrir menu'}
        >
          {mobileOpen ? <CloseIcon /> : <MenuIcon />}
        </MobileMenuTrigger>
      </InnerContainer>
    </StyledHeader>
  );
};

const StyledHeader = styled.header`
  background-color: ${(props) => props.theme.colors.white};
  box-shadow: ${(props) => props.theme.shadows.global};
  position: fixed;
  z-index: 5;
  top: 0;
  left: 0;
  width: 100%;
`;

const InnerContainer = styled.div`
  ${container};
  height: 60px;
  display: flex;
  justify-content: space-around;
  align-items: center;
  ${(props) => props.theme.mq.desktop} {
    height: 67px;
  }
`;

const NavContainer = styled.div`
  display: flex;
  align-items: center;
  flex: 1;
`;

const StyledNav = styled.nav<{ open: boolean }>`
  position: absolute;
  top: 60px;
  width: 100%;
  left: 100%;
  height: calc(100vh - 60px);
  padding: 20px 24px;
  border-top: 1px solid ${(props) => props.theme.colors.border.lightBorder};
  transition: all 0.4s ease-in-out;
  transform: translateX(${(props) => (props.open ? '-100%' : '0')});
  ${(props) => props.theme.mq.desktop} {
    transform: none;
    padding: 0;
    height: auto;
    position: static;
    display: flex;
    align-items: center;
    margin-left: 60px;
    border: 0;
  }
  a {
    text-decoration: none;
  }
  & > * {
    margin-right: 22px;
  }
`;

const DesktopAvatar = styled.div`
  display: none;
  a {
    text-decoration: none;
  }
  ${(props) => props.theme.mq.desktop} {
    display: block;
  }
`;

const MobileMenuTrigger = styled.button`
  border: 0;
  display: flex;
  align-items: center;
  justify-content: center;
  background: transparent;
  padding: 0;
  ${(props) => props.theme.mq.desktop} {
    display: none;
  }
  svg {
    color: ${(props) => props.theme.colors.text.globalText};
  }
`;

Header.defaultProps = {
  // eslint-disable-next-line react/display-name
  navItemLinkWrapper: (props, children) => (
    <a href={props.href} key={props.href}>
      {children}
    </a>
  ),
  // eslint-disable-next-line react/display-name
  avatarLinkWrapper: (children) => <a href="#">{children}</a>,
};
