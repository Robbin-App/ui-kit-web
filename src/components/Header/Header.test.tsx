import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import { baseRobbinTheme } from '../../styles';
import { defaultProps } from './Header.stories';
import { Header } from './index';

const renderPropComponent = (theme) => (
  <ThemeProvider theme={theme}>
    <Header {...defaultProps} />
  </ThemeProvider>
);

describe('Header', () => {
  test('renders', () => {
    const render = renderer
      .create(renderPropComponent(baseRobbinTheme))
      .toJSON();
    expect(render).toMatchSnapshot();
  });
});
