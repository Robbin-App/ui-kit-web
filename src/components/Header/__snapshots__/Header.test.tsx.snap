// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders 1`] = `
<header
  className="sc-gtsrHT kAhoGx"
>
  <div
    className="sc-dlnjwi iGgBrV"
  >
    <svg
      fill="none"
      height="27"
      viewBox="0 0 113 27"
      width="113"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M21.76 20.68C21.68 20.9333 21.5533 21.12 21.38 21.24C21.2067 21.36 21.0133 21.42 20.8 21.42C20.5333 21.42 20.2933 21.3267 20.08 21.14C19.88 20.9667 19.78 20.74 19.78 20.46C19.78 20.34 19.8 20.22 19.84 20.1L24.48 5.56C24.56 5.30667 24.6867 5.12 24.86 5C25.0333 4.88 25.2267 4.82 25.44 4.82C25.7067 4.82 25.94 4.91333 26.14 5.1C26.3533 5.27333 26.46 5.5 26.46 5.78C26.46 5.9 26.44 6.02 26.4 6.14L21.76 20.68ZM32.4542 18.26C33.1209 18.3 33.4542 18.6 33.4542 19.16C33.4542 19.48 33.3209 19.7267 33.0542 19.9C32.8009 20.06 32.4342 20.1267 31.9542 20.1L31.4142 20.06C29.1742 19.9 28.0542 18.7 28.0542 16.46V12.1H27.0542C26.6942 12.1 26.4142 12.02 26.2142 11.86C26.0276 11.7 25.9342 11.4667 25.9342 11.16C25.9342 10.8533 26.0276 10.62 26.2142 10.46C26.4142 10.3 26.6942 10.22 27.0542 10.22H28.0542V8.38C28.0542 8.02 28.1676 7.73333 28.3942 7.52C28.6209 7.30667 28.9276 7.2 29.3142 7.2C29.6876 7.2 29.9876 7.30667 30.2142 7.52C30.4409 7.73333 30.5542 8.02 30.5542 8.38V10.22H32.2542C32.6142 10.22 32.8876 10.3 33.0742 10.46C33.2742 10.62 33.3742 10.8533 33.3742 11.16C33.3742 11.4667 33.2742 11.7 33.0742 11.86C32.8876 12.02 32.6142 12.1 32.2542 12.1H30.5542V16.64C30.5542 17.6267 31.0076 18.1533 31.9142 18.22L32.4542 18.26ZM38.997 9.98C40.357 9.98 41.3637 10.32 42.017 11C42.6837 11.6667 43.017 12.6867 43.017 14.06V18.94C43.017 19.3 42.9104 19.5867 42.697 19.8C42.4837 20 42.1904 20.1 41.817 20.1C41.4704 20.1 41.1837 19.9933 40.957 19.78C40.7437 19.5667 40.637 19.2867 40.637 18.94V18.5C40.4104 19.02 40.0504 19.4267 39.557 19.72C39.077 20.0133 38.517 20.16 37.877 20.16C37.2237 20.16 36.6304 20.0267 36.097 19.76C35.5637 19.4933 35.1437 19.1267 34.837 18.66C34.5304 18.1933 34.377 17.6733 34.377 17.1C34.377 16.38 34.557 15.8133 34.917 15.4C35.2904 14.9867 35.8904 14.6867 36.717 14.5C37.5437 14.3133 38.6837 14.22 40.137 14.22H40.637V13.76C40.637 13.1067 40.497 12.6333 40.217 12.34C39.937 12.0333 39.4837 11.88 38.857 11.88C38.4704 11.88 38.077 11.94 37.677 12.06C37.277 12.1667 36.8037 12.3267 36.257 12.54C35.9104 12.7133 35.657 12.8 35.497 12.8C35.257 12.8 35.057 12.7133 34.897 12.54C34.7504 12.3667 34.677 12.14 34.677 11.86C34.677 11.6333 34.7304 11.44 34.837 11.28C34.957 11.1067 35.1504 10.9467 35.417 10.8C35.8837 10.5467 36.437 10.3467 37.077 10.2C37.7304 10.0533 38.3704 9.98 38.997 9.98ZM38.377 18.36C39.0437 18.36 39.5837 18.14 39.997 17.7C40.4237 17.2467 40.637 16.6667 40.637 15.96V15.54H40.277C39.3837 15.54 38.6904 15.58 38.197 15.66C37.7037 15.74 37.3504 15.88 37.137 16.08C36.9237 16.28 36.817 16.5533 36.817 16.9C36.817 17.3267 36.9637 17.68 37.257 17.96C37.5637 18.2267 37.937 18.36 38.377 18.36ZM46.7978 20.12C46.4378 20.12 46.1378 20.02 45.8978 19.82C45.6711 19.6067 45.5578 19.3067 45.5578 18.92V6.98C45.5578 6.59333 45.6711 6.3 45.8978 6.1C46.1378 5.9 46.4378 5.8 46.7978 5.8C47.1578 5.8 47.4578 5.9 47.6978 6.1C47.9378 6.3 48.0578 6.59333 48.0578 6.98V18.92C48.0578 19.3067 47.9378 19.6067 47.6978 19.82C47.4578 20.02 47.1578 20.12 46.7978 20.12ZM58.3759 17.32C58.6026 17.32 58.7826 17.4067 58.9159 17.58C59.0626 17.7533 59.1359 17.9867 59.1359 18.28C59.1359 18.6933 58.8893 19.04 58.3959 19.32C57.9426 19.5733 57.4293 19.78 56.8559 19.94C56.2826 20.0867 55.7359 20.16 55.2159 20.16C53.6426 20.16 52.3959 19.7067 51.4759 18.8C50.5559 17.8933 50.0959 16.6533 50.0959 15.08C50.0959 14.08 50.2959 13.1933 50.6959 12.42C51.0959 11.6467 51.6559 11.0467 52.3759 10.62C53.1093 10.1933 53.9359 9.98 54.8559 9.98C55.7359 9.98 56.5026 10.1733 57.1559 10.56C57.8093 10.9467 58.3159 11.4933 58.6759 12.2C59.0359 12.9067 59.2159 13.74 59.2159 14.7C59.2159 15.2733 58.9626 15.56 58.4559 15.56H52.5559C52.6359 16.48 52.8959 17.16 53.3359 17.6C53.7759 18.0267 54.4159 18.24 55.2559 18.24C55.6826 18.24 56.0559 18.1867 56.3759 18.08C56.7093 17.9733 57.0826 17.8267 57.4959 17.64C57.8959 17.4267 58.1893 17.32 58.3759 17.32ZM54.9159 11.74C54.2359 11.74 53.6893 11.9533 53.2759 12.38C52.8759 12.8067 52.6359 13.42 52.5559 14.22H57.0759C57.0493 13.4067 56.8493 12.7933 56.4759 12.38C56.1026 11.9533 55.5826 11.74 54.9159 11.74ZM66.9409 9.98C68.1143 9.98 68.9876 10.3133 69.5609 10.98C70.1343 11.6467 70.4209 12.6533 70.4209 14V18.92C70.4209 19.2933 70.3076 19.5867 70.0809 19.8C69.8676 20.0133 69.5676 20.12 69.1809 20.12C68.7943 20.12 68.4876 20.0133 68.2609 19.8C68.0343 19.5867 67.9209 19.2933 67.9209 18.92V14.14C67.9209 13.38 67.7743 12.8267 67.4809 12.48C67.2009 12.1333 66.7543 11.96 66.1409 11.96C65.4209 11.96 64.8409 12.1867 64.4009 12.64C63.9743 13.0933 63.7609 13.7 63.7609 14.46V18.92C63.7609 19.2933 63.6476 19.5867 63.4209 19.8C63.1943 20.0133 62.8876 20.12 62.5009 20.12C62.1143 20.12 61.8076 20.0133 61.5809 19.8C61.3676 19.5867 61.2609 19.2933 61.2609 18.92V11.18C61.2609 10.8333 61.3743 10.5533 61.6009 10.34C61.8276 10.1267 62.1343 10.02 62.5209 10.02C62.8676 10.02 63.1476 10.1267 63.3609 10.34C63.5876 10.54 63.7009 10.8067 63.7009 11.14V11.68C64.0343 11.1333 64.4809 10.7133 65.0409 10.42C65.6009 10.1267 66.2343 9.98 66.9409 9.98ZM78.0402 18.26C78.7068 18.3 79.0402 18.6 79.0402 19.16C79.0402 19.48 78.9068 19.7267 78.6402 19.9C78.3868 20.06 78.0202 20.1267 77.5402 20.1L77.0002 20.06C74.7602 19.9 73.6402 18.7 73.6402 16.46V12.1H72.6402C72.2802 12.1 72.0002 12.02 71.8002 11.86C71.6135 11.7 71.5202 11.4667 71.5202 11.16C71.5202 10.8533 71.6135 10.62 71.8002 10.46C72.0002 10.3 72.2802 10.22 72.6402 10.22H73.6402V8.38C73.6402 8.02 73.7535 7.73333 73.9802 7.52C74.2068 7.30667 74.5135 7.2 74.9002 7.2C75.2735 7.2 75.5735 7.30667 75.8002 7.52C76.0268 7.73333 76.1402 8.02 76.1402 8.38V10.22H77.8402C78.2002 10.22 78.4735 10.3 78.6602 10.46C78.8602 10.62 78.9602 10.8533 78.9602 11.16C78.9602 11.4667 78.8602 11.7 78.6602 11.86C78.4735 12.02 78.2002 12.1 77.8402 12.1H76.1402V16.64C76.1402 17.6267 76.5935 18.1533 77.5002 18.22L78.0402 18.26ZM84.803 20.16C83.803 20.16 82.923 19.9533 82.163 19.54C81.4163 19.1267 80.8363 18.5333 80.423 17.76C80.0096 16.9867 79.803 16.0867 79.803 15.06C79.803 14.0333 80.0096 13.14 80.423 12.38C80.8363 11.6067 81.4163 11.0133 82.163 10.6C82.923 10.1867 83.803 9.98 84.803 9.98C85.803 9.98 86.6763 10.1867 87.423 10.6C88.183 11.0133 88.763 11.6067 89.163 12.38C89.5763 13.14 89.783 14.0333 89.783 15.06C89.783 16.0867 89.5763 16.9867 89.163 17.76C88.763 18.5333 88.183 19.1267 87.423 19.54C86.6763 19.9533 85.803 20.16 84.803 20.16ZM84.783 18.22C85.5963 18.22 86.2163 17.9533 86.643 17.42C87.0696 16.8867 87.283 16.1 87.283 15.06C87.283 14.0333 87.0696 13.2533 86.643 12.72C86.2163 12.1733 85.603 11.9 84.803 11.9C84.003 11.9 83.383 12.1733 82.943 12.72C82.5163 13.2533 82.303 14.0333 82.303 15.06C82.303 16.1 82.5163 16.8867 82.943 17.42C83.3696 17.9533 83.983 18.22 84.783 18.22ZM95.3869 20.16C93.9335 20.16 92.7602 19.8733 91.8669 19.3C91.4402 19.0467 91.2269 18.6867 91.2269 18.22C91.2269 17.9533 91.3002 17.74 91.4469 17.58C91.5935 17.4067 91.7735 17.32 91.9869 17.32C92.2135 17.32 92.5269 17.4267 92.9269 17.64C93.3535 17.84 93.7402 17.9933 94.0869 18.1C94.4469 18.2067 94.9002 18.26 95.4469 18.26C96.0069 18.26 96.4402 18.1667 96.7469 17.98C97.0669 17.7933 97.2269 17.5333 97.2269 17.2C97.2269 16.9733 97.1602 16.7933 97.0269 16.66C96.9069 16.5267 96.6802 16.4067 96.3469 16.3C96.0269 16.18 95.5335 16.0467 94.8669 15.9C93.6802 15.66 92.8269 15.32 92.3069 14.88C91.8002 14.44 91.5469 13.8333 91.5469 13.06C91.5469 12.4733 91.7202 11.9467 92.0669 11.48C92.4135 11 92.8869 10.6333 93.4869 10.38C94.1002 10.1133 94.7935 9.98 95.5669 9.98C96.1269 9.98 96.6669 10.0533 97.1869 10.2C97.7069 10.3333 98.1735 10.5333 98.5869 10.8C99.0135 11.0667 99.2269 11.42 99.2269 11.86C99.2269 12.1267 99.1535 12.3533 99.0069 12.54C98.8602 12.7133 98.6802 12.8 98.4669 12.8C98.3202 12.8 98.1735 12.7733 98.0269 12.72C97.8935 12.6533 97.7202 12.56 97.5069 12.44C97.1202 12.24 96.7869 12.0933 96.5069 12C96.2269 11.8933 95.8802 11.84 95.4669 11.84C94.9735 11.84 94.5802 11.94 94.2869 12.14C94.0069 12.34 93.8669 12.6133 93.8669 12.96C93.8669 13.2933 94.0135 13.5467 94.3069 13.72C94.6002 13.8933 95.1669 14.0667 96.0069 14.24C96.9002 14.4267 97.6002 14.64 98.1069 14.88C98.6135 15.12 98.9735 15.42 99.1869 15.78C99.4135 16.14 99.5269 16.6 99.5269 17.16C99.5269 18.0667 99.1469 18.7933 98.3869 19.34C97.6402 19.8867 96.6402 20.16 95.3869 20.16Z"
        fill="#2E2E2E"
      />
      <path
        clipRule="evenodd"
        d="M14.9955 3.13282L18.4222 17.5854C18.6178 18.4104 18.4257 19.2798 17.9006 19.9453C17.3756 20.6107 16.5753 20.9993 15.7282 21H2.77043C1.92305 20.9998 1.12234 20.6114 0.596958 19.9459C0.0715709 19.2803 -0.120751 18.4107 0.0749078 17.5854L3.50432 3.13282C3.79764 1.88064 4.91505 0.996495 6.19985 1.00001H12.298C13.5831 0.997018 14.7007 1.88074 14.9955 3.13282ZM6.19991 2.67403C5.69035 2.67276 5.247 3.02289 5.12965 3.51927L3.96574 8.42313C5.28331 8.75181 7.91538 9.57372 10.6862 11.3648C11.0409 11.3048 11.7588 11.0997 12.4086 10.4109C13.3768 9.38316 13.8534 7.60888 13.7846 5.2771L13.3683 3.51927C13.2511 3.02281 12.8077 2.67261 12.2981 2.67403H6.19991ZM15.7278 19.3283H2.77002C2.43409 19.3277 2.11683 19.1735 1.90866 18.9096C1.7005 18.6457 1.62425 18.301 1.70167 17.9737L3.57959 10.0559C4.94723 10.4041 7.95542 11.3568 10.8734 13.531C13.0021 15.1181 14.6977 17.0596 15.925 19.3088C15.8599 19.3203 15.7939 19.3268 15.7278 19.3283ZM13.584 11.6029C13.2207 11.9818 12.7983 12.2989 12.3333 12.5419C14.0044 13.8519 15.4503 15.4266 16.614 17.2035L14.8197 9.63768C14.5145 10.4086 14.1025 11.0636 13.584 11.6029Z"
        fill="#15456F"
        fillRule="evenodd"
      />
    </svg>
    <div
      className="sc-hKFxyN hyibRh"
    >
      <nav
        className="sc-eCApnc jFNQhW"
        open={false}
      >
        <a
          href="/"
        >
          <div
            className="sc-bdnxRM dWkoyX"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M14 6V4h-4v2h4zM4 8v11h16V8H4zm16-2c1.11 0 2 .89 2 2v11c0 1.11-.89 2-2 2H4c-1.11 0-2-.89-2-2l.01-11c0-1.11.88-2 1.99-2h4V4c0-1.11.89-2 2-2h4c1.11 0 2 .89 2 2v2h4z"
                fillRule="evenodd"
              />
            </svg>
            <span>
              Oportunidades
            </span>
          </div>
        </a>
        <a
          href="/applications"
        >
          <div
            className="sc-bdnxRM gwnTur"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M8 16h8v2H8zm0-4h8v2H8zm6-10H6c-1.1 0-2 .9-2 2v16c0 1.1.89 2 1.99 2H18c1.1 0 2-.9 2-2V8l-6-6zm4 18H6V4h7v5h5v11z"
              />
            </svg>
            <span>
              Aplicaciones
            </span>
          </div>
        </a>
        <a
          href="/notifications"
        >
          <div
            className="sc-bdnxRM gwnTur"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 22c1.1 0 2-.9 2-2h-4c0 1.1.9 2 2 2zm6-6v-5c0-3.07-1.63-5.64-4.5-6.32V4c0-.83-.67-1.5-1.5-1.5s-1.5.67-1.5 1.5v.68C7.64 5.36 6 7.92 6 11v5l-2 2v1h16v-1l-2-2zm-2 1H8v-6c0-2.48 1.51-4.5 4-4.5s4 2.02 4 4.5v6z"
              />
            </svg>
            <span>
              Notificaciones
            </span>
          </div>
        </a>
        <a
          href="/edit-profile"
        >
          <div
            className="sc-bdnxRM gwnTur"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 22c1.1 0 2-.9 2-2h-4c0 1.1.9 2 2 2zm6-6v-5c0-3.07-1.63-5.64-4.5-6.32V4c0-.83-.67-1.5-1.5-1.5s-1.5.67-1.5 1.5v.68C7.64 5.36 6 7.92 6 11v5l-2 2v1h16v-1l-2-2zm-2 1H8v-6c0-2.48 1.51-4.5 4-4.5s4 2.02 4 4.5v6z"
              />
            </svg>
            <span>
              Editar Perfil
            </span>
          </div>
        </a>
      </nav>
    </div>
    <div
      className="sc-jSFjdj CpWou"
    >
      <a
        href="#"
      >
        <div
          className="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
        >
          JD
        </div>
      </a>
    </div>
    <button
      aria-hidden={false}
      aria-label="abrir menu"
      className="sc-gKAaRy ekpQdd"
      onClick={[Function]}
      tabIndex={-1}
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
        />
      </svg>
    </button>
  </div>
</header>
`;
